### PR TITLE
feat: cds segment

### PR DESCRIPTION
**Pull request by wozjac**
### Prerequisites

- [x] I have read and understood the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes/features)

### Description

A segment for cds command from [SAP Cloud Application Programming Model - SAP CAP](https://cap.cloud.sap/docs/tools/#command-line-interface-cli). 
In the project root directory you can have .cdsrc.json (or .cdsrc-private.json) file, but it is not mandatory, so package.json dependency is also checked. *.cds files are located in `db', `srv` subdirectories (usually scaffolded with `cds init`).

Best Regards
Jacek

### CODE REVIEW

File: src/segments/cds.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+import (
+	"encoding/json"
+	"oh-my-posh/environment"
+	"oh-my-posh/properties"
+)
+
+type Cds struct {
+	language
+	HasCDSDependency bool
```
Comment (JanDeDobbeleer) [inline src/segments/cds.go]: As the context is already CDS, does it makes sense to rename this to `HasDependency`?

File: src/segments/cds.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	dependencies := map[string]json.RawMessage{}
+
+	if err := json.Unmarshal(objmap["dependencies"], &dependencies); err != nil {
+		return
+	}
+
+	for d := range dependencies {
+		if d == "@sap/cds" {
+			c.HasCDSDependency = true
+			return
```
Comment (JanDeDobbeleer) [inline src/segments/cds.go]: should this be `break`?

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

File: src/segments/cds.go (hunk 1 above)
Comment (wozjac) [reply src/segments/cds.go]: sounds ok

File: src/segments/cds.go (hunk 2 above)
Comment (wozjac) [reply src/segments/cds.go]: will correct

File: src/segments/cds.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+import (
+	"encoding/json"
+	"oh-my-posh/environment"
+	"oh-my-posh/properties"
+)
+
+type Cds struct {
+	language
+	HasDependency bool
```
Comment (JanDeDobbeleer) [inline src/segments/cds.go:11]: This isn't present in the docs template properties it seems

Comment (wozjac) [reply src/segments/cds.go:11]: Could you point to similar case in existing segments? Not sure if I understand correctly.

File: docs/docs/segment-cds.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+## Template Properties
+
+- `.Full`: `string` - the full version
+- `.Major`: `string` - major number
+- `.Minor`: `string` - minor number
+- `.Patch`: `string` - patch number
+- `.Prerelease`: `string` - prerelease info text
+- `.BuildMetadata`: `string` - build metadata
+- `.Error`: `string` - when fetching the version string errors
```
Comment (JanDeDobbeleer) [inline docs/docs/segment-cds.md:55]: @wozjac I would expect to see `HasDependency` also listed here.

File: src/segments/cds.go (hunk 3 above)
Comment (JanDeDobbeleer) [reply src/segments/cds.go:11]: Added a comment above.

**Review by JanDeDobbeleer — APPROVED** [review]

**Pull request merged**